### PR TITLE
Make epoch and best_valid_map not global

### PR DESCRIPTION
**Pull request by benjamin-cates**

### CODE REVIEW

File: pyha_analyzer/train.py [hunk 1]
```diff
@@ -282,25 +287,27 @@ def main(in_sweep=True) -> None:
...
-
-    for _ in range(cfg.epochs):
-        logger.info("Epoch %d", EPOCH)
-
-        train(model_for_run, train_dataloader, val_dataloader, optimizer, scheduler)
-        EPOCH += 1
-        valid_map = valid( model_for_run, val_dataloader, EPOCH)
-        logger.info("Best validation map: %f", BEST_VALID_MAP)
+
+    best_valid_map = 0.0
```
Comment (Sean1572) [inline pyha_analyzer/train.py:342]: reset epoch here as well

**Review by Sean1572 — CHANGES_REQUESTED** [review]
Line 290 in train.py, add epoch=0

File: pyha_analyzer/train.py (hunk 1 above)
Comment (benjamin-cates) [reply pyha_analyzer/train.py:342]: Epoch is now defined as a variable in the for loop so it doesn't need to be reset

**Comment by Sean1572** [comment]
Fixed some merge conflicts, please review

**Review by Sean1572 — APPROVED** [review]

**Pull request merged**